corehq/apps/reports/views.py

```diff
@@ -847,20 +846,12 @@ def email_report(request, domain, report_slug, report_type=ProjectReportDispatch
         notes=form.cleaned_data['notes'], once=once
     )[0]
 
+    recipient_emails = set(form.cleaned_data['recipient_emails'])
     if form.cleaned_data['send_to_owner']:
-        email = request.couch_user.get_email()
-        body = render_full_report_notification(request, content).content
-
-        send_html_email_async.delay(
-            subject, email, body,
-            email_from=settings.DEFAULT_FROM_EMAIL)
-
-    if form.cleaned_data['recipient_emails']:
-        for recipient in form.cleaned_data['recipient_emails']:
-            body = render_full_report_notification(request, content).content
-            send_html_email_async.delay(
-                subject, recipient, body,
-                email_from=settings.DEFAULT_FROM_EMAIL)
+        recipient_emails.add(request.couch_user.get_email())
+
+    for recipient in recipient_emails:
```

Comment:
This creates a new task for each recipient which in the worst case, will attempt to send the report, then render the report as an excel file, then send the email again. If there are a lot of recipients and a large report, this could easily clog the main queue by occupying all the workers, and it repeats a lot of effort. It may be better to pass all recipients at once to `send_email_report`, and then if any of them are too large, send the rest as download links. It would be even better if you modified `export_all_rows_task` to take a list of recipients so that it could render the report once, and then send the rest of the emails. That way you could avoid replicating work, and this whole task would be executed as a single task by one worker instead of spread across all of them